src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -470,7 +656,15 @@ def _validate_up_args(cmd, source, image, repo, registry_server):
         if registry_name and len(registry_name) > MAXIMUM_SECRET_LENGTH:
             raise ValidationError(f"--registry-server ACR name must be less than {MAXIMUM_SECRET_LENGTH} "
                                   "characters when using --repo")
-
+    if custom_location_id or connected_cluster_id:
+        register_provider_if_needed(cmd, CUSTOM_LOCATION_RP)
+        register_provider_if_needed(cmd, KUBERNETES_CONFIGURATION_RP)
+        if location:
+            _ensure_location_allowed(cmd, location, CONTAINER_APPS_RP, "connectedEnvironments")
```

Comment:
Move this logic into validate_environment_location?

Reply:
Need to ensure location allowed before creating custom location and environment.
So the verification logic needs to be placed in the _validate_up_args.